04_hex_and_bytes.md

```diff
@@ -1,32 +1,88 @@
+If the receiver expects to see data in a different order from the sender, it will misinterpret that data.
+Thus, communication protocols specify which order bytes will be sent.
+In the Bitcoin protocol, bytes are transmitted in little-endian order.
+That means that the least significant byte is sent first.
+
+When looking at decimal numbers, humans in the West typically read it from left to right with the most significant digit first to the least significant digit last.
+For example, in the number, `201`, the most significant digit is `2` and the least significant digit is `1`.
+But in the case of this transaction version data, the least significant byte will appear first as we just saw.
+This is a common cause of confusion when learning the Bitcoin protocol: bytes seem to be written in reverse in many parts of the system.
+
+So, the version 1 field will be sent in the following order: `01` followed by three `00`s.
+`01` is the least significant byte and is sent first.
+When doing base math to convert these bytes to integers, we can canceptualize it in reverse so that the `01` is in the least significant position.
+In other words, we can think of it as `00 00 00 01` and then do our normal base 16 math to convert this to an integer.
 
-When looking at decimal numbers, we typically read it from left to right with the most sigificant digit first to the least significant digit last. For example, in the number, `201`, the most significant digit is `2` and the least significant digit is `1`. But in the case of this transaction version data, the least significant byte appears first. So we'd have to take this into account when doing base math. One way to do this is to simply reverse the order of the bytes and then do the base math we're familiar with. For example `01 00 00 00` can be reversed and become `00 00 00 01`. Now, we can do our normal base math which is `0* 16^7 + ... + 1 * 16^0`. We don't need to write all of it out since the other values will just be zero. Therefore the version number is just `1 * 16^0` which is `1`.
+Now, we can do our normal base math which is `0* 16^7 + ... + 1 * 16^0`.
+We don't need to write all of it out since the other values will just be zero.
+Therefore the version number is just `1 * 16^0` which is `1`.
 
-We can also convert each byte to its binary representation and then do some math to convert the binary number to a decimal number. For example `01` in hexadecimal is `00000001` in binary (8 bits). Remember, binary is just a number with base 2. So we can do base 2 math. `0 * 2^7 + ... + 1 * 2^0`. This also equals 1.
+We can also convert each byte to its binary representation and then do some math to convert the binary number to a decimal number.
+For example `01` in hexadecimal is `00000001` in binary (8 bits).
+Remember, binary is just a number with base 2.
+So we can do base 2 math.
+`0 * 2^7 + ...
++ 1 * 2^0`.
```

Comment:
This shows up as a bullet point on the next line in the file preview. Just wanted to check if that was what you intended. Otherwise, this looks chapter looks great. Thanks for all the edits! https://github.com/sb1752/bitcoiner-intro-to-rust/blob/c16c39f5283dc69b24d738904fcfb3ecfbe69e5b/04_hex_and_bytes.md

Reply:
Oops, my bad.